lib/attr_encrypted/adapters/active_record.rb

```diff
@@ -75,6 +75,20 @@ def attr_encrypted(*attrs)
             end
 
             define_method("#{attr}_with_dirtiness=") do |value|
+              ##
+              # In ActiveRecord 5.2+, due to changes to the way virtual
+              # attributes are handled, @attributes[attr].value is nil which
+              # breaks attribute_was. Setting it here returns us to the expected
+              # behavior.
+              if ::ActiveRecord::VERSION::STRING >= "5.2"
+                # This is needed support attribute_was before a record has
+                # been saved
+                set_attribute_was(attr, __send__(attr)) if value != __send__(attr)
```

Comment:
Works well with ActiveRecord 5.2.
However, in ActiveRecord 6.0.0, private method `set_attribute_was` was removed. https://github.com/rails/rails/commit/6b0a9de906f2f96a847b5eb93755db0911f2615d

Reply:
I'm not sure how we should proceed with a fix that only works in a single release. My first thought was to allow this to merge, and for Rails 6 we create a new major release. That will also give us the opportunity to finally deprecate some old code and potentially do some major cleanup of the test suite.

Any thoughts on how we can move forward? Is my suggestion the best course or are there better ideas?

Reply:
Just update the pull request with the code snippet bellow 
https://github.com/attr-encrypted/attr_encrypted/pull/337#issuecomment-658013513